File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/horario/ExcepcionHorarioResponse.java
package pe.albrugroup.schedule_service.entity.response.horario;

import lombok.*;
import pe.albrugroup.schedule_service.entity.enums.TipoExcepcionHorario;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExcepcionHorarioResponse {

    private Long id;
    private LocalDate fecha;
    private TipoExcepcionHorario tipo;
    private LocalTime horaEntrada;
    private LocalTime horaSalida;
    private LocalTime inicioAlmuerzo;
    private LocalTime finAlmuerzo;
    private Boolean laborable;
    private String motivo;
}

File: frontend/src/app/shared/models/schedule/excepcion-horario-response.ts
export interface ExcepcionHorarioResponse {
  id: number;
  fecha: string;
  tipo: string;
  horaEntrada?: string | null;
  horaSalida?: string | null;
  inicioAlmuerzo?: string | null;
  finAlmuerzo?: string | null;
  laborable?: boolean | null;
  motivo: string;
  creadoPor?: number | null;
}
